Add OS and architecture to binary filename in CompileBinaryCommand

kpanuragh opened the pull request:
Updated the CompileBinaryCommand to include the operating system type and architecture in the binary filename. This change ensures that binaries
 are uniquely identified by their build environment, preventing potential conflicts when binaries for different environments are stored in the sa
 directory.

kpanuragh committed:
Add OS and architecture to binary filename in CompileBinaryCommand

 Updated the CompileBinaryCommand to include the operating system type and architecture in the binary filename. This change ensures that binaries
 are uniquely identified by their build environment, preventing potential conflicts when binaries for different environments are stored in the sa
 directory.

kpanuragh committed:
feat: normalize OS type to lowercase in binary compilation

 The OS type is now converted to lowercase to ensure consistency and prevent potential issues with case sensitivity during binary compilation. Th
 change helps maintain uniformity across different environments.

kpanuragh committed:
feat: normalize OS and architecture names for binary compilation

 Updated the CompileBinaryCommand to standardize OS and architecture names. This ensures compatibility and consistency across different
 environments by mapping 'darwin' to 'mac', 'Linux' to 'linux', and 'Windows NT' to 'windows'. Similarly, 'x86_64' is mapped to 'x64' and 'aarch6
 to 'arm64'. This change helps in generating binaries with a consistent naming convention.

kpanuragh commented:
Mappings have also been added to match Node.js OS architecture names and OS type names.

kpanuragh commented:
An update for the OS has also been added: https://github.com/laravel/vs-code-extension/pull/22

eznix86 commented on app/Commands/CompileBinaryCommand.php:
```diff
@@ -20,8 +20,21 @@ public function handle(): void
+            $architecture = 'x64';
+        } elseif ($architecture === 'aarch64') {
+            $architecture = 'arm64';
+        }
```
Related to:
https://github.com/laravel/vs-code-extension/pull/22#discussion_r1887335324

`php_uname('m')` should me mapped:
```
aarch64, alpha, arc, arm, i?86, ia64, m68k, mips, mips64, parisc, ppc, ppc64, ppc64le, ppcle, riscv64, s390, s390x, sh, sparc, sparc64, x86_64
```
same for `php_uname('s')`
```php
$osType = match (strtolower(php_uname('s'))) {
    'darwin' => 'darwin',
    'linux' => 'linux',
    'windows nt' => 'win32',
    default => strtolower(php_uname('s')),
};

$architecture = match (php_uname('m')) {
    'x86_64', 'amd64' => 'x64', 
    'aarch64' => 'arm64', 
    'armv7l', 'armv6l' => 'arm', 
    'i386', 'i686' => 'ia32',
    default => php_uname('m'),
};

$destination = base_path("bin/php-parser-{$osType}-{$architecture}-{$version}");

```
@joetannenbaum

kpanuragh replied:
Updated code base

kpanuragh committed:
Refactor OS and architecture detection in CompileBinaryCommand

 Updated the OS and architecture detection logic using PHP's match expression for improved readability and maintainability. This change simplifie
 the code by replacing multiple if-else statements with a more concise and expressive syntax, making it easier to understand and extend in the
 future.

kpanuragh commented:
@joetannenbaum  I think you fixed the issue do I need to keep the PR

aeadedoyin commented:
@kpanuragh yeah it fixes it, it now works correctly on my intel, have you tested on yours too?

cc @eznix86

joetannenbaum commented:
@kpanuragh thank you for the PR! Ended up handling this a different way, but I appreciate you submitting this, going to close this one.
